signing/signing.go

```diff
@@ -0,0 +1,324 @@
+package signing
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/go-openapi/runtime"
+	"github.com/google/go-github/v42/github"
+	"github.com/ossf/scorecard/v2/cron/data"
+	"github.com/rhysd/actionlint"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	hashedrekord "github.com/sigstore/rekor/pkg/types/hashedrekord/v0.0.1"
+	rekord "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+)
+
+type ScorecardOutput struct {
+	SarifOutput string
+	JsonOutput  string
+}
+
+// verifySignature receives the scorecard analysis payload, looks up its associated tlog entry and
+// certificate, and extracts the repository's workflow file to ensure its legitimacy.
+func VerifySignature(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "error reading http request body", http.StatusInternalServerError)
+		log.Println(err)
+		return
+	}
+
+	// Unmarshal request body.
+	var scorecardOutput ScorecardOutput
+	err = json.Unmarshal(reqBody, &scorecardOutput)
+	if err != nil {
+		http.Error(w, "error unmarshalling request body", http.StatusInternalServerError)
+		log.Println(err)
+		return
+	}
+
+	// Lookup results payload to get the repo info from the corresponding entry & cert.
+	sarifRepoPath, sarifRepoRef, err1 := lookupPayload(ctx, []byte(scorecardOutput.SarifOutput))
```

Comment:
why do we need sarif output? I'm just curious whether we have a use case for it already or not

Reply:
Currently the sarif output is not being used for anything. It is just being verified and uploaded to the GCS bucket in anticipation of future usefulness.

Does the sarif contain any info that's not in the json, or is this redundant?